### DISCUSSION TITLE
Is the internal spa value system Celsius or Fahrenheit?

### DISCUSSION THREAD

**Discussion by WesWesthaver** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#685] [General]
I've been watching my spa pump's display for a while now and it seems that the temperature values are displayed somewhat oddly. 

I have my pump set to display in Fahrenheit and quite often the temperature value will skip a number. Looking at the conversion chart below it would appear that the pump's internal temperature measurements are in Celsius and when converted to Fahrenheit it would skip a number at times:

```
Celsius	Fahrenheit
30	86	(86.000°F)
31	87	(87.800°F)
32	89	(89.600°F)
33	91	(91.400°F)
34	93	(93.200°F)
35	95	(95.000°F)
36	96	(96.800°F)
37	98	(98.600°F)
38	100	(100.40°F)
39	102	(102.20°F)
40	104	(104.00°F)
```
Looking at the temperatures reported to Home Assistant I see that the pump never reports these values: 92,94,96,98, and 101. I'm assuming that the spa pump rounds the temperature values and then displays the results?

I kinda wish that the pump used Fahrenheit internally so that the temperature resolution would be better. My pump seems to have a rather large temperature variation window. I think this would be smaller if the pump dealt internally in Fahrenheit.

@visualapproach are you able to see the temperature as it is reported by the temperature sensor or are you limited to seeing only what is sent to the display panel?

Am I even close to understanding what is going on inside the pump or am I all wet?

**Comment by visualapproach** [comment]
Hi there! The 4-wire pumps use C internally and the display converts it to F. So the ESP can only have resolution of C (1.8 F). On all other pumps the ESP only knows what is displayed. So if you choose F on the pump display, the ESP uses F internally. Except for the virtual temperature, which is C internally, but at single float precision so it doesn't affect the resolution.
